pkg/operator/ceph/disruption/clusterdisruption/pools.go

```diff
@@ -130,7 +130,14 @@ func (r *ReconcileClusterDisruption) getOngoingDrains(request reconcile.Request)
 			nodeList := &corev1.NodeList{}
 			err = r.client.List(context.TODO(), nodeList, client.MatchingLabels{corev1.LabelHostname: nodeHostname})
 			nodeNum := len(nodeList.Items)
-			if err != nil || nodeNum < 1 {
+			if nodeNum < 1 {
```

Comment:
If you don't check for that specific error before checking `nodeNum` you might get an invalid error condition where it exists but there was some other error. Why not first check for `errors.IsNotFound(err)`?

Reply:
I don't think the list operation will return an IsNotFound, just an empty node list.

Reply:
We still need to check for `err != nil` even if the behavior is that the list is empty. There could be some other error than IsNotFound

Reply:
Yes definitely! Don't know how I missed that :grimacing:

Reply:
Turns out I did not miss that, I just left the PR alone so long I grew unfamiliar. It's in the adjacent else-if

Reply:
I still see that we should check `err != nil` before evaluating the `nodeNum`.